tests/instrumentation-framework/src/main/java/org/hawkular/apm/tests/dockerized/environment/DockerComposeExecutor.java

```diff
@@ -0,0 +1,149 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.hawkular.apm.tests.dockerized.environment;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.util.Arrays;
+
+import org.hawkular.apm.tests.dockerized.exception.EnvironmentException;
+import org.hawkular.apm.tests.dockerized.model.TestEnvironment;
+
+/**
+ * @author Pavol Loffay
+ */
+public class DockerComposeExecutor implements TestEnvironmentExecutor {
+    private static final String HAWKULAR_APM_NETWORK = "hawkular-apm";
+
+    private final String apmBindAddress;
+
+
+    /**
+     * @param apmBindAddress Address of default gateway (host OS where is running APM server). This network
+     *                       is also used as default in docker-compose
+     */
+    public DockerComposeExecutor(String apmBindAddress) {
+        this.apmBindAddress = apmBindAddress;
+    }
+
+    /**
+     * Create network which is used as default in docker-compose.yml
+     * This should be run before {@link DockerComposeExecutor#run(TestEnvironment)}
+     */
+    public void createNetwork() {
+        String[] command = new String[] {
+                "docker", "network", "create", "--driver=bridge", "--subnet=" + apmBindAddress + "/24",
+                "--gateway=" + apmBindAddress, HAWKULAR_APM_NETWORK,
+        };
+
+        runShellCommand(command);
+    }
+
+    @Override
+    public String run(TestEnvironment testEnvironment) {
+        String[] command = new String[] {
+            "docker-compose", "-f", testEnvironment.getDockerCompose(), "up", "-d"
+        };
+
+        runShellCommand(command);
+
+        return testEnvironment.getDockerCompose();
+    }
+
+    @Override
+    public void clean(String dockerComposeDirectory) {
+
+        String[] command;
+        try {
+            command = new String[]{
+                    "docker-compose", "-f", dockerComposeDirectory, "down"
+            };
+            runShellCommand(command);
+        } catch (EnvironmentException ex) {
+            System.out.println(ex);
+        }
+
+        command = new String[] {
+            "docker", "network", "rm", HAWKULAR_APM_NETWORK
+        };
+        runShellCommand(command);
+    }
+
+    /**
+     *
+     * @param id Id of the environment, Concretely path to docker-compose file.
+     * @param serviceName Service name in running environment. Service defined in docker-compose.
+     * @param script Script to execute
+     */
+    @Override
+    public void execScript(String id, String serviceName, String script) {
+
+        String scriptDockerLocalDir = "/opt/hawkular-apm-test-local";
```

Comment:
it can be anywhere inside docker container. But calling java would not create that directory inside running docker container.